builtin/providers/azurerm/resource_arm_local_network_gateway.go

```diff
@@ -101,6 +101,9 @@ func resourceArmLocalNetworkGatewayRead(d *schema.ResourceData, meta interface{}
 		return err
 	}
 	name := id.Path["localNetworkGateways"]
+	if name == "" {
+		return fmt.Errorf("Cannot find parameter 'localNetworkGateways' from '%s'", id.Path)
+	}
```

Comment:
Error message seems confusing (Cannot find parameter 'localNetworkGateways' from 'map[localNetworkgateways:lng]').  If you can't find something better, at least change "from" to "in".